README.adoc

```diff
@@ -0,0 +1,402 @@
+  return numbers.reduce((a, b) => a + b)
+}
+
+sum(1, 2, 3)
+```
+
+**Pass**
+
+```js
+const sum (numbers) =>
+  numbers.reduce((a, b) => a + b)
+
+sum([1, 2, 3])
+
+const args = node.arguments
+```
+
+### `no-class`
+
+> Classes are nice tools to use when programming with the object-oriented paradigm, as they hold internal state and give access to methods on the instances. In functional programming, having stateful objects is more harmful than helpful, and should be replaced by the use of pure functions.
+
+Further reading: https://github.com/joshburgess/not-awesome-es6-classes/[Not Awesome: ES6 Classes; A curated list of resources on why ES6 (aka ES2015) classes are NOT awesome]
+
+**Fail**
+
+```js
+class Polygon {
+  constructor (height, width) {
+    this.height = height
+    this.width = width
+  }
+}
+```
+
+**Pass**
+
+```js
+const polygon = (height, width) => ({
+  height: height,
+  width: width,
+})
+```
+
+### `fp/no-delete`
+
+> delete is an operator to remove fields from an object or elements from an array. This purposely mutates data, which is not wanted when doing functional programming.
+
+Further reading: https://github.com/google/google-api-nodejs-client/issues/375[Avoid using delete operator]
+
+**Fail**
+```js
+delete foo
+delete foo.bar
+delete foo[bar]
+```
+**Pass**
+```js
+import F from "ramda"
+
+const fooWithoutBar = F.omit(["bar"], foo)
+const fooWithoutField = F.omit([bar], foo)
+```
+
+### `fp/no-events`
+
+> The use of EventEmitter with the events module provided by Node.js promotes implicit side-effects by emitting and listening to events. Instead of events, you should prefer activating the wanted effects by calling the functions you wish to use explicitly.
+
+Probably what you should do is use a https://gist.github.com/staltz/868e7e9bc2a7b8c1f754[functional reactive programming] library: https://github.com/cujojs/most[`most`], https://github.com/Reactive-Extensions/RxJS[`rxjs`].
+
+**Fail**
+
+```js
+import EventEmitter from "events"
+```
+
+### `fp/no-get-set`
+
+**Fail**
+```js
+const person = {
+  name: 'Some Name',
+  get age () {
+    return this._age
+  },
+  set age (n) {
+    if (n < 0) {
+      this._age = 0
+    } else if (n > 100) {
+      this._age = 100
+    } else {
+      this._age = n
+    }
+  }: 20
+};
+
+person.__defineGetter__("name", function () {
+  return this.name || "John Doe";
+})
+
+person.__defineSetter__("name", function (name) {
+  this.name = name.trim();
+})
+```
+**Pass**
+```js
+import F from "ramda"
+
+const person = {
+  name: "Some Name",
+  age: 20,
+}
+
+const clamp = (n, min, max) =>
+  n <= min ? min :
+  n >= max ? max :
+             n
+
+const setAge = (age, person) =>
+  F.merge(person, { age: clamp(age, 0, 100) })
+```
+
+### `fp/no-let`
+
+> If you want to program as if your variables are immutable, part of the answer is to not allow your variables to be reassigned. By not allowing the use of let and var, variables that you declared may not be reassigned.
```

Comment:
I mean, isn't Vue itself the beneficiary of the fact that immutability finally caught on in mainstream GUI programming via React? Ditto with Vuex via Redux? These are widely hailed breakthroughs. Isn't immutability also the force powering revolution in infrastructure manifested as containers? So, if anything, then, amen? : )

If `const` sucks for you then I don't know what you're going to think about https://facebook.github.io/immutable-js/ : D

Haha seriously though, lets discuss. I want to understand.